# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: ipc/glue/ProtocolUtils.cpp
# Commit: b09ae4c3a94b
# Full Hash: b09ae4c3a94b663279189c50172c1f403d499183
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2021-06-22 21:29:07
# Regressor Bug: 1706374
# File Overlap Count: 1
# Description:
#   Bug 1706374 - Part 11: Add NodeController component bridging IPC and Ports, r=handyman
#   
#   The NodeController and NodeChannel types act as the backbone connecting the
#   existing IPC logic and driving the ports routing code. Individual NodeChannel
#   objects wrap and respond to messages from IPC::Channel, and the NodeController
# ==============================================================================

diff -r 04422175004b -r b09ae4c3a94b ipc/glue/ProtocolUtils.cpp
--- a/ipc/glue/ProtocolUtils.cpp	Mon Jun 21 21:53:10 2021 +0000
+++ b/ipc/glue/ProtocolUtils.cpp	Mon Jun 21 21:53:10 2021 +0000
@@ -639,6 +639,11 @@
   mOtherPid = aOtherPid;
 }
 
+bool IToplevelProtocol::Open(ScopedPort aPort, base::ProcessId aOtherPid) {
+  SetOtherProcessId(aOtherPid);
+  return GetIPCChannel()->Open(std::move(aPort), mSide);
+}
+
 bool IToplevelProtocol::Open(UniquePtr<Transport> aTransport,
                              base::ProcessId aOtherPid, MessageLoop* aThread,
                              mozilla::ipc::Side aSide) {